runner/do.go

```diff
@@ -0,0 +1,70 @@
+package runner
+
+import (
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcdiag/op"
+)
+
+var _ Runner = Do{}
+
+// Do runs shell commands.
+type Do struct {
+	Runners []Runner `json:"runners"`
+	log     hclog.Logger
+}
+
+// NewDo returns a pointer to a new Do runner
+func NewDo(l hclog.Logger, runners []Runner) *Do {
+	return &Do{
+		Runners: runners,
+		log:     l,
+	}
+}
+
+func (d Do) ID() string {
+	return "Do"
+}
+
+// Run asynchronously executes all Runners and returns the resulting ops when the last one has finished
+func (d Do) Run() []op.Op {
+	// sortMap maps each runner-index to its resulting ops, so that we avoid a non-deterministically-ordered
+	// slice that includes ops mixed together from *all* runners
+	sortMap := make(map[int][]op.Op)
+	// mu protects sortMap, which is going to be concurrently accessed by our runner goroutines
+	var mu sync.Mutex
+
+	// Start WaitGroup that will wait for all of our Runners
+	wg := new(sync.WaitGroup)
+	wg.Add(len(d.Runners))
+
+	for i, r := range d.Runners {
+		d.log.Info("running operation", "runner", r.ID())
+		go func(m *sync.Mutex, wg *sync.WaitGroup, sm *map[int][]op.Op, r Runner, ridx int) {
+			// Dereference pointer TODO(is there a better way?)
```

Comment:
Since the map isn't needed until we have the lock, an alternative way (not sure it's better or worse, but it avoids a variable declaration) would be to do something like `(*sm)[ridx] = o` after the lock.